Document qlight CLI options (#212)
ConsenSys/doc.goquorum: docs/configure-and-manage/manage/qlight-node.md

> 

Reviewer: - Maybe change the heading to "Configure a GoQuorum qlight node" and move this page to the "Configure" section. (In general, use simple present tense for verb headings.)
- Link to this how-to page from the qlight concept page.
- Edit the note in lines 10-11: "Setting up a server node to support MPS is not required, for example, when setting up a qlight client for offloading processing from a full node."

Author: Updated

> Configure the qlight client using the following command line options:

Reviewer: Since the options descriptions are simply repeated here, I recommend changing to:

```suggestion
Configure the qlight client using the [`--qlight.client.*`](../../reference/cli-syntax.md#qlightclient) command line options.
```

And deleting the bulleted list.

- [`--qlight.client`](../../reference/cli-syntax.md#qlightclient): This marks the node as a qlight client.
- [`--qlight.client.psi`](../../reference/cli-syntax.md#qlightclientpsi): This specifies the PSI which this client will support (The default is `private`).
- [`--qlight.client.serverNode`](../../reference/cli-syntax.md#qlightclientservernode): The Node ID of the server node.
- [`--qlight.client.serverNodeRPC`](../../reference/cli-syntax.md#qlightclientservernoderpc): The RPC URL of the server node.

> If the server node has the RPC API secured using TLS, then the qlight client requires the following:

Reviewer: Since the options descriptions are simply repeated here, I recommend changing to:

```suggestion
If the server node has the RPC API secured using TLS, then you must set the [`--qlight.client.rpc.tls.*`](../../reference/cli-syntax.md#qlightclientrpctls) command line options.
```

And deleting the bulleted list.

- [`--qlight.client.rpc.tls`](../../reference/cli-syntax.md#qlightclientrpctls): Use TLS when forwarding RPC API calls.
- [`--qlight.client.rpc.tls.insecureskipverify`](../../reference/cli-syntax.md#qlightclientrpctlsinsecureskipverify): Skip TLS verification.
- [`--qlight.client.rpc.tls.cacert`](../../reference/cli-syntax.md#qlightclientrpctlscacert): The qlight client certificate authority.
- [`--qlight.client.rpc.tls.cert`](../../reference/cli-syntax.md#qlightclientrpctlscert): The qlight client certificate.
- [`--qlight.client.rpc.tls.key`](../../reference/cli-syntax.md#qlightclientrpctlskey): The qlight client certificate private key.

If the qlight client node is halted, on restart it resyncs with any blocks that were missed when it was not running.

## Configure server node

> Configure the qlight server using the following command line options:

Reviewer: Since the options descriptions are simply repeated here, I recommend changing to:

```suggestion
Configure the qlight server using the [`--qlight.server.*`](../../reference/cli-syntax.md#qlightserver) command line options.
```

And deleting the bulleted list.

- [`--qlight.server.p2p.port`](../../reference/cli-syntax.md#qlightserverp2pport): The RPC listening port.
- [`--qlight.server.p2p.maxpeers`](../../reference/cli-syntax.md#qlightserverp2pmaxpeers): The maximum number of qlight clients that are supported.

### Network IP restriction

This restricts communication to specified IP networks (CIDR masks).
Specify the network mask on the qlight server using [`--qlight.server.p2p.netrestrict`](../../reference/cli-syntax.md#qlightserverp2pnetrestrict).

### File based permissioning

File based permissioning allows qlight clients to be checked against a permissioned list and a disallowed list.
Enable file based permissioning on the server node using [`--qlight.server.p2p.permissioning`](../../reference/cli-syntax.md#qlightserverp2ppermissioning).

The default files are `permissioned-nodes.json` and `disallowed-nodes.json`.
However, you can specify a file prefix using [`--qlight.server.p2p.permissioning.prefix`](../../reference/cli-syntax.md#qlightserverp2ppermissioningprefix),
in which case the filename is: the prefix, followed by a hyphen, followed by the default file name.

> ## Using the enterprise authorization protocol integration

Reviewer: ```suggestion
## Configure enterprise authorization
```

Use simple present tense for verb headings.

> When using [JSON-RPC security](json-rpc-api-security.md#enterprise-authorization-protocol-integration)

Reviewer: In line 56, specify what "This" is. Also, don't use vague terms like "[leverage](https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/l/leverage)."

E.g.:
```
The [enterprise authorization protocol integration](json-rpc-api-security.md#enterprise-authorization-protocol-integration) only allows authenticated clients to connect to the server.

Author: Fixed

you must provide an access token to communicate to the qlight server.

Enable auth tokens in the qlight client using [`--qlight.client.token.enabled`](../../reference/cli-syntax.md#qlightclienttokenenabled).

Once enabled, specify an initial value using [`--qlight.client.token.value`](../../reference/cli-syntax.md#qlightclienttokenvalue).

Specify a refresh mechanism for the token using [`--qlight.client.token.management`](../../reference/cli-syntax.md#qlightclienttokenmanagement).
The valid values are:

> - `none` - the token is not refreshed (this mechanism is for development/testing purposes only).

Reviewer: I would delete this and following two lines since it is described in the linked reference docs (and worded differently there).

Author: Updated

> You can add an encryption layer on the qlight client-server communication.

Reviewer: Change this heading to "Configure TLS" or similar.

Author: Fixed

> Configure the encryption layer using the following options:

Reviewer: Since the options descriptions are simply repeated here, I recommend changing to:

```suggestion
Configure the encryption layer using the [`qlight.tls.*`](../../reference/cli-syntax.md#qlighttls) command line options.
```

And deleting the bulleted list.
